src/realm/object-store/shared_realm.cpp

```diff
@@ -478,7 +478,11 @@ void Realm::update_schema(Schema schema, uint64_t version, MigrationFunction mig
     });
 
     if (!in_transaction) {
-        transaction().promote_to_write();
+        constexpr bool nonblocking = true;
+        auto success = transaction().promote_to_write(nonblocking);
+        if (!success) {
+            throw LogicError(ErrorCodes::WrongTransactionState, "Could not create write transaction");
```

Comment:
maybe mention it's during a schema update, otherwise the message is pretty vague

Reply:
👍